chore: add PR template (#220)
signalfx/splunk-otel-js-web: .github/pull_request_template.md

# Description

_Replace this with a description of the PR. Explain the problem it solves and the proposed solution._

> _List Github/JIRA issue(s) this PR fixes._

Reviewer: Not sure Jira issues should go here if it's a public repo.

## Type of change

> Please delete options that are not relevant.

Reviewer: ```suggestion
Delete options that are not relevant.
```

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore or internal change (changes not visible to the consumers of the package)

> - This change requires a documentation update

Reviewer: ```suggestion
- Documentation update
```

> # How Has This Been Tested?

Reviewer: ```suggestion
# How has this been tested?
```

> - [ ] Tested manually

Reviewer: ```suggestion
- [ ] Manual testing
```

- [ ] Added unit tests
- [ ] Added integration tests

# Checklist:

- [ ] Changelogs have been updated
- [ ] Unit tests have been added/updated
- [ ] Documentation has been updated

> - [ ] Team's resident editor is tagged on the PR (if there are changes to the documentation)

Reviewer: ```suggestion
- [ ] Team's resident technical writer tagged on the PR (if there are changes to the documentation)
```
